util_linux/mkfs_ext2.rs

```diff
@@ -975,38 +975,38 @@ pub unsafe extern "C" fn mkfs_ext2_main(
   }
   // misc. Values are chosen to match mke2fs 1.41.9
   if ::std::mem::size_of::<u16>() as libc::c_ulong == 4i32 as libc::c_ulong {
-    (*sb).s_state = 1i32 as u32 as u16
+    (*sb).s_state = 1
   } else if ::std::mem::size_of::<u16>() as libc::c_ulong == 2i32 as libc::c_ulong {
-    (*sb).s_state = 1i32 as u16
+    (*sb).s_state = 1
   } else if ::std::mem::size_of::<u16>() as libc::c_ulong == 1i32 as libc::c_ulong {
-    (*sb).s_state = 1i32 as u8 as u16
+    (*sb).s_state = 1
   } else {
     BUG_wrong_field_size(); // TODO: what's 1?
   } // 180 days
   if ::std::mem::size_of::<u32>() as libc::c_ulong == 4i32 as libc::c_ulong {
-    (*sb).s_creator_os = 0 as u32
+    (*sb).s_creator_os = 0
   } else if ::std::mem::size_of::<u32>() as libc::c_ulong == 2i32 as libc::c_ulong {
-    (*sb).s_creator_os = 0 as u16 as u32
+    (*sb).s_creator_os = 0
   } else if ::std::mem::size_of::<u32>() as libc::c_ulong == 1i32 as libc::c_ulong {
-    (*sb).s_creator_os = 0 as u8 as u32
+    (*sb).s_creator_os = 0
   } else {
     BUG_wrong_field_size();
   }
   if ::std::mem::size_of::<u32>() as libc::c_ulong == 4i32 as libc::c_ulong {
-    (*sb).s_checkinterval = (24i32 * 60i32 * 60i32 * 180i32) as u32
+    (*sb).s_checkinterval = (24 * 60 * 60 * 180) as u32
   } else if ::std::mem::size_of::<u32>() as libc::c_ulong == 2i32 as libc::c_ulong {
-    (*sb).s_checkinterval = (24i32 * 60i32 * 60i32 * 180i32) as u16 as u32
+    (*sb).s_checkinterval = std::u16::MAX as u32 // (24 * 60 * 60 * 180) as u16 as u32
```

Comment:
Hmm, there's some weird behaviour I don't understand going on there: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=29c6b3ebb37d81097b304d926057beda

Any idea what's happening there?

Reply:
I think this is why: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1b3ad0a8303f7d96d7dc0baf1d049259. Integer casts truncate in rust which is why they aren't just the `MAX` value: https://doc.rust-lang.org/1.30.0/book/first-edition/casting-between-types.html#numeric-casts.

Reply:
Hmm, I see. What do you think we should do in this case then? Use the truncated value?

Reply:
In this case, yeah we kind of have to. This is exactly what the C version did (oddly), and who knows what could break if we tried to get too clever. It may certainly be worth cleaning these things up in the future but no point in messing with it right now IMHO.

Reply:
👍